File: framework/languages/java/samples/java/SupportChat/Shared/src/main/java/systems/zlink/samples/supportchat/shared/contracts/Messages.java
package systems.zlink.samples.supportchat.shared.contracts;

public final class Messages {
    private Messages() {
    }

    public record AuthenticateReq(String accessToken) {
    }

    public record AuthenticateRes(String actorId, String displayName, String role) {
    }

    public record AuthenticateUserReq(String accessToken) {
    }

    public record AuthenticateUserRes(
        boolean accepted,
        String actorId,
        String displayName,
        String role,
        String reason) {
    }

    public record OpenConversationApiReq(
        String customerActorId,
        String customerDisplayName,
        String subject) {
    }

    public record OpenConversationApiRes(String conversationId, String status) {
    }

    public record AllocateConversationReq(
        String customerActorId,
        String customerDisplayName,
        String subject) {
    }

    public record AllocateConversationRes(String conversationId, String status, ConversationState state) {
    }

    public record ActorRefWire(String nodeRid, String actorId, long generation) {
    }

    public record EnsureSupportUserActorReq(String actorId, String displayName, String role) {
    }

    public record EnsureSupportUserActorRes(String actorId, ActorRefWire actor) {
    }

    public record OpenConversationReq(String subject) {
    }

    public record OpenConversationRes(String conversationId, ConversationState state) {
    }

    public record SetAgentAvailableReq(boolean isAvailable) {
    }

    public record SetAgentAvailableRes(boolean isAvailable) {
    }

    public record JoinConversationReq(String conversationId) {
    }

    public record JoinConversationRes(ConversationState state) {
    }

    public record SendChatMessageReq(String conversationId, String text) {
    }

    public record SendChatMessageRes(ChatMessage message, ConversationState state) {
    }

    public record SetTypingReq(String conversationId, boolean isTyping) {
    }

    public record CloseConversationReq(String conversationId, String reason) {
    }

    public record CloseConversationRes(ConversationState state) {
    }

    public record ParticipantJoinedNotify(
        String conversationId,
        String actorId,
        String role,
        ConversationState state) {
    }

    public record ConversationAssignedNotify(String conversationId, ConversationState state) {
    }

    public record ChatMessageNotify(
        String conversationId,
        ChatMessage message,
        ConversationState state) {
    }

    public record TypingChangedNotify(
        String conversationId,
        String actorId,
        boolean isTyping,
        ConversationState state) {
    }

    public record ConversationIdleNotify(String conversationId, ConversationState state) {
    }

    public record ConversationClosedNotify(String conversationId, ConversationState state) {
    }

    public record ConversationState(
        String conversationId,
        String subject,
        String status,
        String customerActorId,
        String agentActorId,
        long lastMessageSeq,
        Long lastMessageAtUnixMs,
        Long idleDeadlineUnixMs) {
    }

    public record ChatMessage(
        String conversationId,
        long messageSeq,
        String senderActorId,
        String text,
        long sentAtUnixMs) {
    }

    public record JoinConversationSupportReq(String conversationId, String actorId, String displayName, String role) {
    }

    public record SendChatMessageSupportReq(String conversationId, String actorId, String text) {
    }

    public record SetTypingSupportReq(String conversationId, String actorId, boolean isTyping) {
    }

    public record CloseConversationSupportReq(String conversationId, String actorId, String reason) {
    }

    public record ServerAssertionRequest(String conversationId) {
    }

    public record ServerAssertionResponse(boolean passed, String reason) {
    }
}

File: framework/languages/java/samples/kotlin/SupportChat/Shared/src/main/kotlin/systems/zlink/samples/kotlin/supportchat/shared/contracts/Messages.kt
package systems.zlink.samples.kotlin.supportchat.shared.contracts

data class AuthenticateReq(
    val accessToken: String,
)

data class AuthenticateRes(
    val actorId: String,
    val displayName: String,
    val role: String,
)

data class AuthenticateUserReq(
    val accessToken: String,
)

data class AuthenticateUserRes(
    val accepted: Boolean,
    val actorId: String?,
    val displayName: String?,
    val role: String?,
    val reason: String?,
)

data class OpenConversationApiReq(
    val customerActorId: String,
    val customerDisplayName: String,
    val subject: String,
)

data class OpenConversationApiRes(
    val conversationId: String,
    val status: String,
)

data class AllocateConversationReq(
    val customerActorId: String,
    val customerDisplayName: String,
    val subject: String,
)

data class AllocateConversationRes(
    val conversationId: String,
    val status: String,
    val state: ConversationState? = null,
)

data class ActorRefWire(
    val nodeRid: String,
    val actorId: String,
    val generation: Long,
)

data class EnsureSupportUserActorReq(
    val actorId: String,
    val displayName: String,
    val role: String,
    val participantId: String = actorId,
)

data class EnsureSupportUserActorRes(
    val actorId: String,
    val actor: ActorRefWire,
)

data class EnsureAgentConversationReq(
    val rosterActorId: String,
    val displayName: String,
    val conversationId: String,
)

data class EnsureAgentConversationRes(
    val actor: ActorRefWire,
    val state: ConversationState,
)

data class OpenConversationReq(
    val subject: String,
)

data class OpenConversationRes(
    val conversationId: String,
    val state: ConversationState,
)

data class SetAgentAvailableReq(
    val isAvailable: Boolean,
)

data class SetAgentAvailableRes(
    val isAvailable: Boolean,
)

data class JoinConversationReq(
    val conversationId: String? = null,
)

data class JoinConversationRes(
    val state: ConversationState,
)

data class SendChatMessageReq(
    val conversationId: String? = null,
    val text: String,
)

data class SendChatMessageRes(
    val message: ChatMessage,
    val state: ConversationState,
)

data class SetTypingReq(
    val conversationId: String? = null,
    val isTyping: Boolean,
)

data class CloseConversationReq(
    val conversationId: String? = null,
    val reason: String?,
)

data class CloseConversationRes(
    val state: ConversationState,
)

data class ParticipantJoinedNotify(
    val conversationId: String,
    val actorId: String,
    val role: String,
    val state: ConversationState,
)

data class ConversationAssignedNotify(
    val conversationId: String,
    val state: ConversationState,
)

data class ChatMessageNotify(
    val conversationId: String,
    val message: ChatMessage,
    val state: ConversationState,
)

data class TypingChangedNotify(
    val conversationId: String,
    val actorId: String,
    val isTyping: Boolean,
    val state: ConversationState,
)

data class ConversationIdleNotify(
    val conversationId: String,
    val state: ConversationState,
)

data class ConversationClosedNotify(
    val conversationId: String,
    val state: ConversationState,
)

data class ConversationState(
    val conversationId: String,
    val subject: String,
    val status: String,
    val customerActorId: String,
    val agentActorId: String?,
    val lastMessageSeq: Long,
    val lastMessageAtUnixMs: Long?,
    val idleDeadlineUnixMs: Long?,
)

data class ChatMessage(
    val conversationId: String,
    val messageSeq: Long,
    val senderActorId: String,
    val text: String,
    val sentAtUnixMs: Long,
)

data class JoinConversationSupportReq(
    val conversationId: String,
    val actorId: String,
    val displayName: String,
    val role: String,
)

data class SendChatMessageSupportReq(
    val conversationId: String,
    val actorId: String,
    val text: String,
)

data class SetTypingSupportReq(
    val conversationId: String,
    val actorId: String,
    val isTyping: Boolean,
)

data class CloseConversationSupportReq(
    val conversationId: String,
    val actorId: String,
    val reason: String?,
)

data class ServerAssertionRequest(
    val conversationId: String,
)

data class ServerAssertionResponse(
    val passed: Boolean,
    val reason: String?,
)
